### DISCUSSION TITLE
Question about new Visa Secure Data Fields rules

### DISCUSSION THREAD

**Discussion by matejsemancik** [braintree/braintree-android-drop-in#481] [Q&A] [answered]
Hi, we use drop-in SDK in both our Android and iOS apps to collect payment card information.

I was notified by business, that there are new Visa Network changes - [Updates to Visa Secure Data Fields mandate](https://developer.paypal.com/braintree/articles/risk-and-security/compliance/network-updates/2024/v-s24#what-are-the-updates-to-visa-secure-data-fields-mandate) which they want us to integrate into our mobile application.

Is this something that will be automatically handled by Drop-In SDK, or do we need to do something on application-side?

I found related issue with web drop-in but that was not really helpful to me: https://github.com/braintree/braintree-web-drop-in/issues/922

This is how we essentially collect payment card information right now:
```kotlin
dropInClient.launchDropIn(
    DropInRequest().apply {
        this.isGooglePayDisabled = true
        this.isPayPalDisabled = true
    }
)

// And then send payment method nonce to backend on success
```

Thanks!

**Comment by sarahkoop** [comment, marked as answer]
Hi @matejsemancik - Yes you should include the fields listed in the [Updates to Visa Secure Data Fields mandate](https://developer.paypal.com/braintree/articles/risk-and-security/compliance/network-updates/2024/v-s24#what-are-the-updates-to-visa-secure-data-fields-mandate) by passing a custom `ThreeDSecureRequest` to your `DropInRequest` ([Android](https://github.com/braintree/braintree-android-drop-in/blob/main/Drop-In/src/main/java/com/braintreepayments/api/DropInRequest.java#L140)) or `BTThreeDSecureRequest` to your `BTDropInRequest` ([iOS](https://github.com/braintree/braintree-ios-drop-in/blob/main/Sources/BraintreeDropIn/Public/BraintreeDropIn/BTDropInRequest.h#L49)). The fields outlined in the table linked above can be set through the `ThreeDSecureRequest` object. 

For a list of all fields available for customizing the request, see [ThreeDSecureRequest](https://github.com/braintree/braintree_android/blob/main/ThreeDSecure/src/main/java/com/braintreepayments/api/threedsecure/ThreeDSecureRequest.java#L22) or [BTThreeDSecureRequest](https://github.com/braintree/braintree_ios/blob/main/Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift#L8) and specifically the ThreeDSecureAdditionalInformation classes.

**Reply by matejsemancik** [reply]
Thanks for clarification!

**Comment by matejsemancik** [comment]
Hi @sarahkoop an additional question you maybe will know answer to. I wasn't really able to find more additional info for this and description in documentation is not very specific -- do we need to pass a local IP address that can be easily retrieved using system APIs, or do we need to send a public IP address of device obtained by calling some 3rd party service such as https://www.ipify.org/ ?
